### PR TITLE
👌 IMPROVE: allow passing filterDropdownProps to Table columns

### PR DESCRIPTION
**Pull request by ds1371dani**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [x] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 💡 Background and solution

image

Currently when we want to implement filtering and search on Table component, we don't have access to the Dropdown component, I suggest allowing to pass DropdowonProps so we can customize Table filter & search more. For example I need my filters to open above the Table instead of inside it (need to change Dropdown placement), but currently I can't achieve this.

### 📝 Changelog

User has the ability to change any of Dropdown props if needed.

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English | :new: add filterDropdownProps to `Table` columns object |
| 🇨🇳 Chinese | :new: 将filterDropdownProps添加到`Table`列对象 |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

---

### <samp>🤖 Generated by Copilot at 062ad08</samp>

This pull request adds a new feature to the table component that lets users customize the filter dropdown for each column. It adds a new `filterDropdownProps` property to the `ColumnType` interface and passes it to the `FilterDropdown` component.

### <samp>🤖 Generated by Copilot at 062ad08</samp>

* Add `filterDropdownProps` property to column configuration to allow customizing filter dropdown component ([link](https://github.com/ant-design/ant-design/pull/45515/files?diff=unified&w=0#diff-3fee134a410a783dfd5f6f89b1a8f7e7eb1ba8e332aeaf4cafdefc4c723088e9R14), [link](https://github.com/ant-design/ant-design/pull/45515/files?diff=unified&w=0#diff-3fee134a410a783dfd5f6f89b1a8f7e7eb1ba8e332aeaf4cafdefc4c723088e9R135-R138), [link](https://github.com/ant-design/ant-design/pull/45515/files?diff=unified&w=0#diff-af396bfff8e0daa4eb8b9f25ed47a6006b544d36c1733636ad578f7109f9ee5aR527))
  * Use `DropdownProps` type from `dropdown` component to define `filterDropdownProps` type, omitting some properties that are not relevant or handled internally ([link](https://github.com/ant-design/ant-design/pull/45515/files?diff=unified&w=0#diff-3fee134a410a783dfd5f6f89b1a8f7e7eb1ba8e332aeaf4cafdefc4c723088e9R14), [link](https://github.com/ant-design/ant-design/pull/45515/files?diff=unified&w=0#diff-3fee134a410a783dfd5f6f89b1a8f7e7eb1ba8e332aeaf4cafdefc4c723088e9R135-R138))
  * Pass `filterDropdownProps` to `FilterDropdown` component in `components/table/hooks/useFilter/FilterDropdown.tsx` and spread them on the `Dropdown` component ([link](https://github.com/ant-design/ant-design/pull/45515/files?diff=unified&w=0#diff-af396bfff8e0daa4eb8b9f25ed47a6006b544d36c1733636ad578f7109f9ee5aR527))

### CODE REVIEW

File: components/table/index.en-US.md [hunk 1]
```diff
@@ -187,6 +187,7 @@ One of the Table `columns` prop for describing the table's columns, Column has t
 | defaultSortOrder | Default order of sorted values | `ascend` \| `descend` | - |  |
 | ellipsis | The ellipsis cell content, not working with sorter and filters for now.<br />tableLayout would be `fixed` when `ellipsis` is `true` or `{ showTitle?: boolean }` | boolean \| {showTitle?: boolean } | false | showTitle: 4.3.0 |
 | filterDropdown | Customized filter overlay | ReactNode \| (props: [FilterDropdownProps](https://github.com/ant-design/ant-design/blob/ecc54dda839619e921c0ace530408871f0281c2a/components/table/interface.tsx#L79)) => ReactNode | - |  |
+| filterDropdownProps | Customized dropdown props | [DropdownProps](https://ant.design/components/dropdown#api) | - |  |
```
Comment (afc163) [inline components/table/index.en-US.md:190]: How about add original dropdown node to filterDropdown?

```jsx
filterDropdown: (props, originalNode) => {
  return originalNode;
}
```

```jsx
filterDropdown: (props, originalNode) => {
  return React.cloneElement(originalNode, { ...dropdownProps });
}
```

Comment (ds1371dani) [reply components/table/index.en-US.md:190]: It is better since we won't have to add a prop, but I'm not sure about it's type checking. I'll have a look

Comment (ds1371dani) [reply components/table/index.en-US.md:190]: Wow great, i thought this was forgotten. I am still willing to implement it but you guys should decide on the solution.

Personally i like adding **filterDropdownProps** and removing **filterDropdownOpen** prop. but if you guys prefer adding **originalNode** i'm okay with that too, just tell me where to put it for the next major version

**Comment by afc163** [comment]
Supported in https://github.com/ant-design/ant-design/issues/45898

**Pull request closed without merging**